gitlab/http_backends/requests_backend.py

```diff
@@ -10,3 +11,40 @@ def __init__(self, session: Optional[requests.Session] = None) -> None:
     @property
     def client(self) -> requests.Session:
         return self._client
+
+    def http_request(
+        self,
+        method: Union[str, bytes],
+        url: str,
+        json: Optional[Union[Dict[str, Any], bytes]] = None,
```

Comment:
The `bytes` here seems incorrect, I think we need to fix the typing in our `_prepare_send_data` which is probably the reason you added it here. I will have a look.

Also just thinking, since `_prepare_send_data` involves backend-specific code (at least the `MultipartEncoder` is requests/toolbelt specific), I assume we will potentially move (some of) that code into the backend as a follow-up.

Reply:
Thanks @nejch for the review. Do you want to hold this PR before `_prepare_send_data` is amended or we can continue with this one and provide a fix later on?

Reply:
Had a quick look through requests/urllib3/json and seems like `bytes` is totally fine for json as well, although I don't expect it coming from our code. Maybe we can leave it as is, for now. Sorry for the detour here.